crates/aptos-crypto/src/traits.rs

```diff
@@ -74,16 +77,36 @@ pub trait ValidCryptoMaterialStringExt: ValidCryptoMaterial {
     /// When trying to convert from bytes, we simply decode the string into
     /// bytes before checking if we can convert.
     fn from_encoded_string(encoded_str: &str) -> std::result::Result<Self, CryptoMaterialError> {
-        let bytes_out = ::hex::decode(encoded_str);
+        let mut start = 0;
+        if encoded_str.starts_with("0x") {
+            start = 2;
+        }
+
+        let hex_len = encoded_str.len() - start;
+        const ENCODED_KEY_SIZE: usize = PRIVATE_KEY_SIZE * 2;
+        // If the string is too short, pad it
+        let encoded_str = if hex_len < ENCODED_KEY_SIZE {
+            let mut hex_str = String::with_capacity(ENCODED_KEY_SIZE);
+            for _ in 0..ENCODED_KEY_SIZE - hex_len {
```

Comment:
I believe you can pad with 
```
assert_eq!("0000012345", format!("{:0>10}", 12345));
```

Reply:
Can't cause it's a string, there's a long convoluted problem around trait inheritance here.

Reply:
who or what is this for? I'm a little worried about this as it is within the crypto crate where we anticipate things are done ... sensibly.